File: backend/src/backend/models/event.py
from datetime import datetime
from sqlalchemy.orm import Mapped, mapped_column, relationship
from sqlalchemy import Text, SmallInteger, DateTime
from sqlalchemy.sql import func
from backend.db.base_model import Base


class Event(Base):
    __tablename__ = "Event"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)

    name: Mapped[str] = mapped_column(Text, nullable=False)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    quota: Mapped[int | None] = mapped_column(SmallInteger, nullable=True)

    started_at: Mapped[datetime | None] = mapped_column(
        DateTime(timezone=True),
        nullable=True
    )

    ended_at: Mapped[datetime | None] = mapped_column(
        DateTime(timezone=True),
        nullable=True
    )

    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now()
    )

    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now()
    )

    registrations: Mapped[list["Registration"]] = relationship(
        back_populates="event"
    )

File: frontend/types/event.ts
export interface Event {
  id: number;
  name: string;
  description: string | null;
  quota: number | null;
  started_at: string | null;
  ended_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface EventCreateInput {
  name: string;
  description?: string | null;
  quota?: number | null;
  started_at?: string | null;
  ended_at?: string | null;
}

export interface EventUpdateInput {
  name?: string;
  description?: string | null;
  quota?: number | null;
  started_at?: string | null;
  ended_at?: string | null;
}

export interface EventCreateInput {
  name: string;
  description?: string | null;
  quota?: number | null;
  started_at?: string | null;
  ended_at?: string | null;
}

export interface EventUpdateInput {
  name?: string;
  description?: string | null;
  quota?: number | null;
  started_at?: string | null;
  ended_at?: string | null;
}
